### DISCUSSION TITLE
Question related to diverging and converging Supply Paths

### DISCUSSION THREAD

**Discussion by GOllagnon** [frePPLe/frepple#592] [Q&A] [answered]
Dear Forum,

I send this message in continuation of the issue # 591, because I think the model is not in sync. with frePPLe principles and philosophy (also, as well as standard operations management and optimisation guidelines...), the prototype is done in the context of discovery of an APS.

Let me explain it quickly:
* a "pilot/prototype" to model operations in the chemical industry.
* the model is rather complex, with many steps, all steps and operations are planning fine ... except the one circled in red
* that operation is planning not enough material of RM1

[FrePPLe_ModelWithLoop.pdf](https://github.com/frePPLe/frepple/files/11781700/FrePPLe_ModelWithLoop.pdf)

I think I know the root cause, can you please confirm my assumptions:
1. There is a loop in the Supply Path, which is not acceptable
2. When the optimization of Inital Opt and Res. 0 is performed, it does not take into accounts all downstreams requirements, but only one of them (e.g. the upper one, or the middle one), leading to partial requirements satisfaction

=> Is it the way it is working?
=> So, of course, we have to identify real bottlenecks, so that to simplify and have Supply Paths without loops?

Thanks & BR,

Guy.

**Comment by jdetaeye** [comment, marked as answer]
This kind of diverging&converging supply paths are ok. 

Loops in the supply path are indeed not ok, but what you show in the picture isn't a loop.   You get a loop if you follow the material/arrows in one direction and end up at the same operation where you started.

**Reply by GOllagnon** [reply]
Thanks for the explanations. Indeed, here we have diverging and converging supply paths.
